### PR TITLE
 conformance testdata: remove tags and trim spaces

### PR DESCRIPTION
**Pull request by yujuhong**
**What this PR does / why we need it**:
To detect whether conformance tests have been changed, the test
argument with the `ConformanceIt` is recorded to conformance.txt.
This change remove additional tags (e.g., [Feature:]) and trim spaces,
so that the detection is less-prone to noise.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by yujuhong** [comment]
Background: I'd like to add tags to some of the conformance tests without triggering a change in the conformance test data.

/cc @dchen1107

File: test/conformance/walk.go [hunk 1]
```diff
@@ -197,6 +198,19 @@ func (v *visitor) getDescription(value string) string {
 		" " + strings.Trim(value, "\"")
 }
 
+var (
+	regexTag = regexp.MustCompile(`(\[[a-zA-Z0-9:-]+\])`)
+)
+
+func processValue(s string) string {
```
Comment (mml) [inline test/conformance/walk.go]: Maybe call this `normalizeTestName`?

**Comment by mml** [comment]
As long as no objections on the conformance list, this LG otherwise.  Thanks!

File: test/conformance/walk.go (hunk 1 above)
Comment (yujuhong) [reply test/conformance/walk.go]: Done.

**Comment by yujuhong** [comment]
> As long as no objections on the conformance list, this LG otherwise. Thanks!

Just sent out the email. Will wait for folks to respond. Thanks for reviewing!

/cc @bgrant0607 @smarterclayton for `testdata` changes

**Comment by yujuhong** [comment]
FYI, email to the cncf-kubernetes-conformance mailing list: https://lists.cncf.io/g/cncf-k8s-conformance/message/414

**Comment by bgrant0607** [comment]
@yujuhong The current impact of this looks like it's just removing trailing spaces?

Are there any automated consumers of conformance.txt that this might impact?

**Comment by yujuhong** [comment]
> @yujuhong The current impact of this looks like it's just removing trailing spaces?

It also removed the `[Flaky]` and `[Slow]` tags. I uploaded an incomplete version before. Updated the PR. 

> Are there any automated consumers of conformance.txt that this might impact?

I don't know who are the consumers of this file. That's part of the reason why I sent out the email to the mailing list. Could you point me to any existing consumers? Thanks!

**Comment by duglin** [comment]
ping @brahmaroutu

**Comment by bgrant0607** [comment]
cc @jagosan

**Comment by bgrant0607** [comment]
Flaky tests shouldn't be part of conformance. #62161

**Comment by bgrant0607** [comment]
Does https://github.com/kubernetes/kubernetes/blob/master/test/conformance/conformance_test.sh run as presubmit?

Does anything else consume conformance.txt?

cc @timothysc

**Comment by bgrant0607** [comment]
Otherwise the change looks fine to me. Thanks.

**Comment by yujuhong** [comment]
Added a unit test for the function. Nothing else was changed.

> Does https://github.com/kubernetes/kubernetes/blob/master/test/conformance/conformance_test.sh run as presubmit?

Yes, that's how I caught the issue.

**Comment by yujuhong** [comment]
> Does anything else consume conformance.txt?

FWIW, a simple search in https://github.com/heptio/sonobuoy returned nothing.

**Pull request merged**